pkg/authentication/handlers.go

```diff
@@ -53,26 +76,57 @@ type API struct {
 	apiKey           string
 	payloadValidator validation.PayloadValidatorInterface
 	oauth2           OAuth2ContextInterface
+	helper           OAuth2HelperInterface
+	authCookiesTTL   time.Duration
 
-	tracer   trace.Tracer
-	logger   logging.LoggerInterface
-	nonceTTL time.Duration
+	tracer        trace.Tracer
+	logger        logging.LoggerInterface
+	cookieManager AuthCookieManagerInterface
 }
 
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	mux.Get("/api/v0/login", a.handleLogin)
+	mux.Get("/api/v0/auth/callback", a.handleCallback)
 }
 
 func (a *API) handleLogin(w http.ResponseWriter, r *http.Request) {
 	// add the Otel HTTP Client
 	r = r.WithContext(OtelHTTPClientContext(r.Context()))
 
-	code := r.URL.Query().Get("code")
+	nonce := a.helper.RandomURLString()
+	state := a.helper.RandomURLString()
+
+	a.cookieManager.SetNonceCookie(w, nonce, a.authCookiesTTL)
+	a.cookieManager.SetStateCookie(w, state, a.authCookiesTTL)
+
+	a.oauth2.LoginRedirect(w, r, nonce, state)
```

Comment:
not sure u need to pass them (`nonce` and `state`) in as u just wrote them in the response, can we grab them from there?

Reply:
I believe this function doesn't have to "know" that those values are in there alreadyI wouldn't like that, also I prefer to keep it direct by passing them explicitly